src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -364,11 +376,219 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
+            out ArrayBuilder<LocalSymbol>? temps,
+            out BitVector positionsAssignedToTemp,
+            out bool receiverIsArgumentSideEffectSequence)
+        {
+            Debug.Assert(argumentRefKindsOpt.IsDefault || argumentRefKindsOpt.Length == arguments.Length);
+            var requiresInstanceReceiver = methodOrIndexer.RequiresInstanceReceiver() && methodOrIndexer is not MethodSymbol { MethodKind: MethodKind.Constructor };
+            Debug.Assert(!requiresInstanceReceiver || rewrittenReceiver != null || _inExpressionLambda);
+            temps = null;
+            positionsAssignedToTemp = BitVector.Null;
+            receiverIsArgumentSideEffectSequence = false;
+
+            if (arguments.IsEmpty)
+            {
+                return arguments;
+            }
+
+            var visitedArgumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(arguments.Length);
+            ArrayBuilder<BoundExpression>? sideEffectsBuilder = null;
+            int lastIndexStored = -1;
+            bool receiverStored = false;
+            ArrayBuilder<RefKind>? refKindsBuilder = null;
+            var parameters = methodOrIndexer.GetParameters();
+
+            for (int i = 0; i < arguments.Length; i++)
+            {
+                InterpolatedStringHandlerData? nullableInterpolatedData = setupInterpolationDataIfNeeded(
+                    i,
+                    argumentRefKindsOpt,
+                    ref receiverStored,
+                    ref lastIndexStored,
+                    ref rewrittenReceiver,
+                    ref temps,
+                    ref sideEffectsBuilder,
+                    ref positionsAssignedToTemp);
+
+                visitedArgumentsBuilder.Add(VisitExpression(arguments[i]));
+
+                if (nullableInterpolatedData is not null)
+                {
+                    var interpolationData = nullableInterpolatedData.GetValueOrDefault();
+                    foreach (var placeholder in interpolationData.ArgumentPlaceholders)
+                    {
+                        RemovePlaceholderReplacement(placeholder);
+                    }
+                }
+            }
+
+            if (temps != null)
+            {
+                Debug.Assert(temps.Count > 0);
+                Debug.Assert(sideEffectsBuilder!.Count > 0);
+                Debug.Assert(!positionsAssignedToTemp.IsNull);
+                Debug.Assert(receiverStored == requiresInstanceReceiver);
+
+                // If this was an interpolated string conversion that needed a receiver, the
+                // first store is actually the receiver, not the argument, so turn the receiver
+                // into a sequence. Otherwise, turn the first actual argument into that sequence.
+
+                var sideEffectsResult = requiresInstanceReceiver ? rewrittenReceiver : visitedArgumentsBuilder[0];
+                Debug.Assert(sideEffectsResult != null);
+
+                var sideEffectsSequence = _factory.Sequence(
```

Comment:
>sideEffectsSequence [](http://example.com/codeflow?start=20&length=19)

It looks like this patch up wouldn't be necessary if we were capturing sideffects in place.